web/src/components/MaskedChanges.res

```diff
@@ -0,0 +1,115 @@
+// A module to manage the masked changes
+
+type t = MonoIndexedDB.MaskedChangeSchema.t
+module Table = Dexie.Table.MakeTable(MonoIndexedDB.MaskedChangeSchema)
+
+let getAll: Dexie.Database.t => Promise.t<array<t>> = dexie => {
+  dexie->Table.findByCriteria({"ctype": #Change})->Dexie.Collection.toArray
+}
+
+let remove = (dexie: Dexie.Database.t, key: string) => dexie->Table.delete(key)->ignore
+
+type changeStatus = Masked | Unmasked
+type status<'a> = (changeStatus, 'a)
+type changeArray = array<status<SearchTypes.change>>
+type change = status<SearchTypes.change>
+type dispatch = (SearchTypes.change => unit, SearchTypes.change => unit)
+
+// Take a dexie db and a change,
+// and return a promise with the masked status.
+let getStatus: (Dexie.Database.t, SearchTypes.change) => Promise.t<status<SearchTypes.change>> = (
+  dexie,
+  change,
+) =>
+  dexie
+  ->Table.getById(change.change_id)
+  ->Promise.then(maskedM =>
+    (
+      switch maskedM {
+      | Some(_) => Masked
+      | _ => Unmasked
+      },
+      change,
+    )->Promise.resolve
+  )
+
+let simpleGetStatus: (changeArray, SearchTypes.change) => changeStatus = (xs, change) => {
+  let maskedM =
+    Belt.Array.keep(xs, ((_, change')) => change.change_id == change'.change_id)->Belt.Array.get(0)
+  switch maskedM {
+  | Some((maskedStatus, _)) => maskedStatus
+  | _ => Unmasked
+  }
+}
+
+// Take a list of change, a dexie db,
+// Then annotate each change,
+let removeMaskedFromArray: (
```

Comment:
I propose to name that function *getStatusFromChanges* (like in the PinnedChanges module). The current name is misleading as nothing is removed from the array.